addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8800,6 +8916,78 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
         colors = Object.assign(defaultColors, colors);
         return pick(colors, ...defaultKeys);
     },
+    /**
+     * Returns the computed colors for the currently selected Connections shape.
+     *
+     * The color is computed based on the bgcolor of the next snippet or on the bgcolor
+     * of the previous if the shape flip on Y.
+     *
+     * @private
+     * @param {String} shape identifier of the selected shape.
+     * @param {Element} $target Element for which the color has to be computed.
+     */
+    _getConnectionsColors(shape, $target = this.$target[0]) {
+        if (!shape.includes("web_editor/Connections/")) {
+            return {};
+        }
+        const defaultColor = this._getShapeDefaultColors(shape);
+        const defaultKey = Object.keys(defaultColor);
+        let nextElement = undefined;
+        const isFlippedY = this._getShapeData($target).flip.includes("y");
+        const isInFooter = $target.closest("footer");
+        const hasPreviousSibling = !!$target.previousElementSibling;
```

Comment:
To check if elementFromPoint + closest could better but not required.

`elementFromPoint(x, y).closest(':has(> .o_we_shape)')` ?